pkg/runtime/runtime_test.go

```diff
@@ -123,3 +124,64 @@ func TestProcessEventTwoCustomTransforms(t *testing.T) {
 		t.Fatal("transform2 should have been called")
 	}
 }
+func TestProcessEventThreeCustomTransformsOneFail(t *testing.T) {
+	// Event from device 1
+	eventIn := models.Event{
+		Device: devID1,
+	}
+	context := excontext.Context{}
+	transform1WasCalled := false
+	transform2WasCalled := false
+	transform3WasCalled := false
+
+	transform1 := func(edgexcontext excontext.Context, params ...interface{}) (bool, interface{}) {
+		transform1WasCalled = true
+		if len(params) != 1 {
+			t.Fatal("should have been passed the first event from CoreData")
+		}
+		if result, ok := params[0].(*models.Event); ok {
+			if ok == false {
+				t.Fatal("Should have receieved CoreData event")
+			}
+
+			if result.Device != devID1 {
+				t.Fatal("Did not receive expected CoreData event")
+			}
+		}
+
+		return false, errors.New("Transform1Result")
+	}
+	transform2 := func(edgexcontext excontext.Context, params ...interface{}) (bool, interface{}) {
+		transform2WasCalled = true
+
+		if params[0] != "Transform1Result" {
+			t.Fatal("Did not recieve result from previous transform")
+		}
+		return true, "Hello"
+	}
+	transform3 := func(edgexcontext excontext.Context, params ...interface{}) (bool, interface{}) {
+		transform3WasCalled = true
+
+		if params[0] != "Transform1Result" {
+			t.Fatal("Did not recieve result from previous transform")
+		}
+		return true, "Hello"
+	}
+	runtime := GolangRuntime{
+		Transforms: []func(edgexcontext excontext.Context, params ...interface{}) (bool, interface{}){transform1, transform2, transform3},
+	}
+
+	result := runtime.ProcessEvent(context, eventIn)
+	if result != nil {
```

Comment:
EdgeX uses the stretchr/testify package which provide assert like in C#. :-)
One down side you have to cal t.Fatal if the assert fail as it is calling t.Fail :-(

	if !assert.NoError(t, err) {
		t.Fatal()
	}

Reply:
I'll bring this in in the next round of testing i do